libs/blocks/preflight/panels/performance.js

```diff
@@ -0,0 +1,277 @@
+};
+// TODO MEP/Personalization
+// TODO mobile / tablet?
+// TODO create ticket for PSI API
+// TODO link to documentation directly within the sections
+const text = {
+  lcpEl: {
+    key: 'lcpEl',
+    title: 'Valid LCP',
+    passed: { description: 'Valid LCP in the first section detected.' },
+    failed: { description: 'No LCP image or video in the first section detected.' },
+  },
+  singleBlock: {
+    key: 'singleBlock',
+    title: 'Single Block',
+    passed: { description: 'First section has exactly one block.' },
+    failed: { description: 'First section has more than one block.' },
+  },
+  imageSize: {
+    key: 'imageSize',
+    title: 'Images size',
+    empty: { description: 'No image as LCP element.' },
+    passed: { description: 'LCP image is less than 100KB.' },
+    failed: { description: 'LCP image is over 100KB.' },
+  },
+  videoPoster: {
+    key: 'videoPoster',
+    title: 'Videos',
+    empty: { description: 'No video as LCP element.' },
+    passed: { description: 'LCP video has a poster attribute' },
+    failed: { description: 'LCP video has no poster attribute.' },
+  },
+  fragments: {
+    key: 'fragments',
+    title: 'Fragments',
+    passed: { description: 'No fragments used within the LCP section.' },
+    failed: { description: 'Fragments used within the LCP section.' },
+  },
+  personalization: {
+    key: 'personalization',
+    title: 'Personalization',
+    passed: { description: 'Personalization is currently not enabled.' },
+    failed: { description: 'MEP or Target enabled.' },
+  },
+  placeholders: {
+    key: 'placeholders',
+    title: 'Placeholders',
+    passed: { description: 'No placeholders found within the LCP section.' },
+    failed: { description: 'Placeholders found within the LCP section.' },
+  },
+  icons: {
+    key: 'icons',
+    title: 'Icons',
+    passed: { description: 'No icons found within the LCP section.' },
+    failed: { description: 'Icons found within the LCP section.' },
+  },
+};
+
+export const config = {
+  items: signal([
+    ...Object.values(text).map(({ key, title }) => ({
+      key,
+      title,
+      icon: icons.empty,
+      description: 'Loading...',
+    })),
+  ]),
+  lcp: null,
+  cls: 0,
+};
+
+export const findItem = (key) => config.items.value.find((item) => item.key === key);
+export const updateItem = ({ key, ...updates }) => {
+  const { items } = config;
+  items.value = items.value.map((item) => (item.key === key ? { ...item, ...updates } : item));
+};
+
+export function conditionalItemUpdate({ emptyWhen, failsWhen, key }) {
+  if (emptyWhen) {
+    updateItem({
+      key,
+      icon: icons.empty,
+      description: text[key].empty.description,
+    });
+    return;
+  }
+  if (failsWhen) {
+    updateItem({
+      key,
+      icon: icons.fail,
+      description: text[key].failed.description,
+    });
+    return;
+  }
+  updateItem({
+    key,
+    icon: icons.pass,
+    description: text[key].passed.description,
+  });
```

Comment:
This feels a bit repetitive. Could we go for something like
```suggestion
  const icon = (emptyWhen && icons.empty) || (failsWhen && icons.fail) || icons.pass;
  const descriptionKey = (emptyWhen && 'empty') || (failsWhen && 'failed') || 'passed';
  updateItem({
    key,
    icon,
    description: text[key][descriptionKey].description,
  });
```

Haven't tested this, some changes might be needed.

Reply:
Cool suggestion, will add in the next commit 👍